src/parser_utils.js

```diff
@@ -1,55 +1,71 @@
 /**
  * Returns all the elements of the given node which nodeName match the given name.
- * @param  {any} node - The node to use to find the matches.
- * @param  {any} name - The name to look for.
- * @return {Array}
+ * @param  {Node} node - The node to use to find the matches.
+ * @param  {String} name - The name to look for.
+ * @return {Array<ChildNode>}
  */
 function childrenByName(node, name) {
-  return [...node.childNodes].filter(
-    (child) =>
+  const children = [];
+  for (const childKey in node.childNodes) {
+    const child = node.childNodes[childKey];
+
+    if (
       child.nodeName === name ||
       name === `vmap:${child.nodeName}` ||
       child.nodeName === `vmap:${name}`
-  );
+    ) {
+      children.push(child);
+    }
+  }
+  return children;
 }
 
 /**
  * Parses a node value giving priority to CDATA as a JSON over text, if CDATA is not a valid JSON it is converted to text
- * @param  {Object} node - The node to parse the value from.
+ * @param  {Node} node - The node to parse the value from.
  * @return {String|Object}
  */
 function parseNodeValue(node) {
-  const childNodes = node && node.childNodes && [...node.childNodes];
-  if (!childNodes) {
+  if (!node || !node.childNodes) {
     return {};
   }
 
   // Trying to find and parse CDATA as JSON
-  const cdatas = childNodes.filter((childNode) => childNode.nodeName === '#cdata-section');
+  const cdatas = [];
+  for (const childKey in node.childNodes) {
```

Comment:
```suggestion
  const childNodes = node.childNodes;
  for (const childKey in childNodes) {
```

As `node.childNodes` is used sevral time in the code shoud use local var, because accessing with `node.childNodes` could be bad in performance in a big VMAP.

Reply:
fixed